### ISSUE TITLE
2.0 bug with lookup of jinja2 templates

### ISSUE THREAD

**Issue by modonnel** [ansible/ansible#13579]
The following line works in 1.9 but fails in 2.0

```
      debug: msg="{{ lookup('template', 'templates/create_body.j2') }}"
```

```
$ cat templates/create_body.j2
{ 'classification': '{{ env }}', 'host_name': '{{ hostname }}' }
```

Error message

```
TASK [debug msg={{ lookup('template', 'templates/create_body.j2') }}] **********
fatal: [localhost]: FAILED! => {"failed": true, "msg": "ERROR! an unexpected type error occurred. Error was sequence item 0: expected string, dict found"}
```

It works if you put quotes around the data structure but this is only a test case.  My real use case is in the uri module with body_format: json.   This doesn't work if the json body is surrounded in quotes. 

thanks,
Mike

**Comment by bcoca** [comment]
so this is probably the 'smart' typing in 2.0, there should be 2 ways around this:

`debug: msg=" {{ lookup('template', 'templates/create_body.j2') }}"`
^ prepend a space to avoid it being detected as a dictionary/hash (begins with "{" )

or use to_json/to_nice_json functions, which will force the type to a json string:

`debug: msg="{{ lookup('template', 'templates/create_body.j2')|to_json }}"`

**Comment by modonnel** [comment]
Neither of those options worked.

modonnel@appl303-stc:~/ansible/playbooks/v2 $ ansible-playbook test_template.yaml

PLAY [test template lookups] ***************************************************

TASK [debug msg= {{ lookup('template', 'templates/create_body.j2') }}] *********
fatal: [localhost]: FAILED! => {"failed": true, "msg": "ERROR! an unexpected type error occurred. Error was sequence item 0: expected string, dict found"}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1

modonnel@appl303-stc:~/ansible/playbooks/v2 $ ansible-playbook test_template.yaml

PLAY [test template lookups] ***************************************************

TASK [debug msg= {{ lookup('template', 'templates/create_body.j2')|to_json }}] *
fatal: [localhost]: FAILED! => {"failed": true, "msg": "ERROR! an unexpected type error occurred. Error was sequence item 0: expected string, dict found"}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1

modonnel@appl303-stc:~/ansible/playbooks/v2 $

**Comment by bcoca** [comment]
looked at it again, you can avoid conversion by adding `convert_data=False`:

```
 - debug: msg="{{ lookup('template', 'templates/create_body.j2', convert_data=False) }}"
```

**Comment by jimi-c** [comment]
Yes, this appears to be a dupe of #11641, which was resolved by adding the above option in ecae636160ca1c81e9c2151264ba754a8bed6a9e.

**Comment by modonnel** [comment]
convert_data=False fixed it.   Thanks for the help.
